lib/matplotlib/axes/_axes.py

```diff
@@ -3284,6 +3284,22 @@ def errorbar(self, x, y, yerr=None, xerr=None,
         if len(x) != len(y):
             raise ValueError("'x' and 'y' must have the same size")
 
+        def check_if_negative(array):
+            try:
+                if np.any(array < 0):
+                    return True
+            except TypeError:   # Don't fail on 'datetime.*' types
+                pass
+
+        if xerr is not None and check_if_negative(xerr):
+            if yerr is not None and check_if_negative(yerr):
+                raise ValueError(
+                    "'xerr' and 'yerr' must have positive numbers")
+            else:
+                raise ValueError("'xerr' must have positive numbers")
+        if check_if_negative(yerr):
+            raise ValueError("'yerr' must have positive numbers")
+
```

Comment:
Maybe just shorten this?  I'm not sure there is a lot of value added in specifying which array is wrong...

```suggestion
        def check_if_negative(array):
            if array is None:
                return False
            try:
                if np.any(array < 0):
                    return True
            except TypeError:   # Don't fail on 'datetime.*' types
                pass

        if check_if_negative(xerr) or check_if_negative(yerr):
            raise ValueError("'xerr' and 'yerr' must have positive numbers")

```

Reply:
yeah, thanks @jklymak agree with you. Let me also fix the tests

Reply:
@jklymak are you OK if I click "Resolve conversation" to keep the conversation page clean, or what is the convention?

Reply:
Trivial stuff can be resolved, or if you took a suggestion at face value...